### DISCUSSION TITLE
Is there any way to set global defaults?

### DISCUSSION THREAD

**Discussion by IllustratedMan-code** [quarto-dev/quarto-cli#8850] [Q&A] [answered]
### Description

I want to change the defaults for pdf export and or globally install custom formats. Is there any way to do this? I am aware that projects exist, but want a "default" solution for all documents.

**Comment by mcanouil** [comment]
What do you mean by "global default"?

Did you look at the "default" project type? <https://quarto.org/docs/projects/quarto-projects.html>

**Comment by IllustratedMan-code** [comment]
I did look at the "default" project type, which is just a barebones project right?

I want a way to set a config file (say in ~/.config/quarto/config.yml) that will act upon every document. Quarto does have defaults, say if I run `quarto render myfile.qmd --to pdf` it creates a pdf with certain metadata settings. I would like a way to change those "global" defaults.

**Reply by mcanouil** [reply]
Quarto works at a project (directory) level by design.
You won't have a system/user-wide in Quarto.

If you want this, then makes hard or symbolic links to a `_quarto.yml` file stored wherever you want.

**Reply by IllustratedMan-code** [reply]
Is that a hard philosophical decision?  Say if I made a pull request that added the functionality I want to quarto, would it be rejected?

**Reply by cscheid** [reply, marked as answer]
Yes, it is, and yes, it would.

With that said, you can also create a project type or format template that matches what you want, and then simply use it when you want to. For us, reproducibility (especially as content moves from being rendered locally to rendered, say, on CI) is a central concern, and ambient user-wide configurations like this tend to cause problems.

**Reply by mcanouil** [reply]
This would make a Quarto project not self-contained and no longer reproducible since others wouldn't have your "config".

I cannot speak for the developers, but what you want clearly goes against the https://quarto.org/ home page.

> Analyze. Share. Reproduce. You have a story to tell with data—tell it with Quarto.

**Reply by IllustratedMan-code** [reply]
Thanks for the help!

**Reply by allefeld** [reply]
I understand the reasoning. However, there are many use-cases for Quarto which do not fit into the self-contained reproducible project mold.

For example, I use Quarto to write solutions for exercises in a module I teach. In my module, there are different contents for each week, and when the module repeats the next year there may be changes and therefore new versions. Each solution fits into a single `qmd` file – it's not a project and doesn't intend to be. Now I can put all my settings into the YAML metadata block, but that means I have to copy-and-paste for every `qmd` file I create. And maybe at some point I decide to change something, and then I have to go back to every file and change it there. I had hoped that custom formats would solve this, but then files get copied into each weekly / yearly folder, cluttering them up. And when I change the format, I have to run `quarto update` everywhere.

I get that your main use case is self-contained projects, so I completely understand that global configuration is not something you want to center and promote. But I don't see that it would hurt as a side-function accommodating other uses like mine?

**Reply by allefeld** [reply]
Maybe as a compromise: Make "installing" an extension something that can be achieved by a YAML metadata key. E.g.

```yaml
---
title: "Presentation"
subtitle: "LexConf 2022"
author: "Your Name"
date: today
use-extension: lexcorp/lexconf
format: lexconf-revealjs
---
```

The extension / custom format would be hosted on GitHub etc., if needed with version management. The contents are not downloaded into the directory of the document, but cached by Quarto.

I think that would satisfy both your reproducibility focus and my wish for light-weight use of custom formats. I.e. the user could decide to just use the generic reference `lexcorp/lexconf` and always get the latest version, or ensure reproducibility with something like `lexcorp/lexconf@v1.2`.
